controllers/share.go

```diff
@@ -0,0 +1,185 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/go-logr/logr"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/k8smetadata/pkg/annotation"
+
+	"github.com/giantswarm/aws-network-topology-operator/pkg/aws"
+	"github.com/giantswarm/aws-network-topology-operator/pkg/util/annotations"
+)
+
+const FinalizerResourceShare = "network-topology.finalizers.giantswarm.io/share"
+
+//counterfeiter:generate . RAMClient
+type RAMClient interface {
+	ApplyResourceShare(context.Context, aws.ResourceShare) error
+	DeleteResourceShare(context.Context, string) error
+}
+
+type ShareReconciler struct {
+	ramClient     RAMClient
+	clusterClient ClusterClient
+}
+
+func NewShareReconciler(clusterClient ClusterClient, ramClient RAMClient) *ShareReconciler {
+	return &ShareReconciler{
+		ramClient:     ramClient,
+		clusterClient: clusterClient,
+	}
+}
+
+func (r *ShareReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&capi.Cluster{}).
+		Complete(r)
+}
+
+func (r *ShareReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.getLogger(ctx)
+
+	logger.Info("Reconciling")
+	defer logger.Info("Done reconciling")
+
+	cluster, err := r.clusterClient.Get(ctx, req.NamespacedName)
+	if k8serrors.IsNotFound(err) {
+		logger.Info("cluster no longer exists")
+		return ctrl.Result{}, nil
+	}
+	if err != nil {
+		logger.Error(err, "failed to get cluster")
+		return ctrl.Result{}, err
+	}
```

Comment:
[You can use `IgnoreNotFound` instead](https://github.com/giantswarm/aws-resolver-rules-operator/blob/main/controllers/resolver_rules.go#L81-L83)